src/main/java/org/apache/mesos/scheduler/plan/DefaultInstallStrategy.java

```diff
@@ -18,12 +19,12 @@ public DefaultInstallStrategy(Phase phase) {
     }
 
     @Override
-    public Optional<Block> getCurrentBlock() {
+    public Optional<Block> getCurrentBlock(Collection<String> dirtiedAssets) {
         if (interrupted.get()) {
             return Optional.empty();
         } else {
             for (Block block : phase.getBlocks()) {
-                if (!block.isComplete()) {
+                if (!block.isComplete() && !dirtiedAssets.contains(block.getName())) {
```

Comment:
Validate that it's non-null, or that it's non-empty?  We don't do null checks anywhere else in the code (and rightly so, it would be too verbose).  And it's valid for `dirtiedAssets` to be empty.